Preferences dialog does not open from file menu

Scene Builder has a central settings (Preferences) dialog, which is reachable from File menu.

### Expected Behavior
When navigating to `File` menu and selecting the menu item `Preferences`, one expects the preferences dialog to open.

### Current Behavior
After selecting the `Preferences` menu item, no window or dialog appears.
There is no error message.

### Steps to Reproduce
* Clone Scene Builder
* Checkout commit ID 810650d
* Build Scene Builder
* Run Scene Builder using Maven or from within IDE
* Navigate to file menu, select `Preferences` menu item

### Your Environment
* Windows 10 22H2
* JDK 23
* OpenJFX 23
* Eclipse 2024-09
* Maven 3.9.9

### First analysis

The class `com.oracle.javafx.scenebuilder.kit.editor.panel.util.AbstractFxmlWindowController` handles the way the window is loaded and opened then. The method `makeRoot()` handles the FXML loading and also any occuring errors. Unfortunately when `Preferences` is selected, a `RuntimeException` is thrown by its `controllerDidLoadFxml()` method. Also unfortunately, there is no useful logging at that point, hence the exception is not properly stated in log file or console log.

```java
    @Override
    protected void makeRoot() {
        final FXMLLoader loader = new FXMLLoader();

        loader.setController(this);
        loader.setLocation(fxmlURL);
        loader.setResources(resources);
        try {
            setRoot((Region)loader.load());
            controllerDidLoadFxml();
        } catch (RuntimeException | IOException x) {
            System.out.println("loader.getController()=" + loader.getController());
            System.out.println("loader.getLocation()=" + loader.getLocation());
            throw new RuntimeException("Failed to load " + fxmlURL.getFile(), x); //NOI18N
        }
    }
```

The problem occurs in `com.oracle.javafx.scenebuilder.app.preferences.PreferencesWindowController` in line 210 where the `PaintPicker` is created within the `controllerDidLoadFxml()` method.

```java
        // PaintPicker delegate shared by alignmentColorPicker and parentRingColorPicker
        final PaintPicker.Delegate delegate = new PaintPickerDelegate();

        // Alignment guides color
        final Color alignmentColor = recordGlobal.getAlignmentGuidesColor();
        alignmentColorPicker = new PaintPicker(delegate, Mode.COLOR);  // The error occurs here!
```

The exception text states, that there might be an issue in line 41 of `/kit/src/main/resources/com/oracle/javafx/scenebuilder/kit/util/control/paintpicker/PaintPicker.fxml`. 

[PaintPicker.fxml#L41](https://github.com/gluonhq/scenebuilder/blob/810650d65770e133e50470418133a1dd7ca738ec/kit/src/main/resources/com/oracle/javafx/scenebuilder/kit/util/control/paintpicker/PaintPicker.fxml#L41)

### Screenshots

![grafik](https://github.com/user-attachments/assets/2c9cd1d0-34b0-48e0-9568-cbc6bead95da)

For the logging part, a more convenient way to get logging without building Scene Builder, is just adding a console logger to the app/resources/logging.properties:
```
- handlers= java.util.logging.FileHandler
+ handlers= java.util.logging.FileHandler, java.util.logging.ConsoleHandler
```
the you'll get the stacktraces of any runtime exception, like this one:
```
Sept 28, 2024 9:04:29 PM com.oracle.javafx.scenebuilder.app.SceneBuilderApp$SceneBuilderUncaughtExceptionHandler uncaughtException
SEVERE: An exception was thrown:
java.lang.RuntimeException: Failed to load scenebuilder/app/target/classes/com/oracle/javafx/scenebuilder/app/preferences/Preferences.fxml
        at com.gluonhq.scenebuilder.kit@24.0.0-SNAPSHOT/com.oracle.javafx.scenebuilder.kit.editor.panel.util.AbstractFxmlWindowController.makeRoot(AbstractFxmlWindowController.java:115)
        at com.gluonhq.scenebuilder.kit@24.0.0-SNAPSHOT/com.oracle.javafx.scenebuilder.kit.editor.panel.util.AbstractWindowController.getRoot(AbstractWindowController.java:87)
        at com.gluonhq.scenebuilder.kit@24.0.0-SNAPSHOT/com.oracle.javafx.scenebuilder.kit.editor.panel.util.AbstractWindowController.getScene(AbstractWindowController.java:107)
        at com.gluonhq.scenebuilder.kit@24.0.0-SNAPSHOT/com.oracle.javafx.scenebuilder.kit.editor.panel.util.AbstractWindowController.getStage(AbstractWindowController.java:130)
        at com.gluonhq.scenebuilder.kit@24.0.0-SNAPSHOT/com.oracle.javafx.scenebuilder.kit.editor.panel.util.AbstractWindowController.openWindow(AbstractWindowController.java:152)
        at com.gluonhq.scenebuilder.app@24.0.0-SNAPSHOT/com.oracle.javafx.scenebuilder.app.SceneBuilderApp.performControlAction(SceneBuilderApp.java:222)
        at com.gluonhq.scenebuilder.app@24.0.0-SNAPSHOT/com.oracle.javafx.scenebuilder.app.menubar.MenuBarController$ApplicationControlActionController.perform(MenuBarController.java:1825)
        at com.gluonhq.scenebuilder.app@24.0.0-SNAPSHOT/com.oracle.javafx.scenebuilder.app.menubar.MenuBarController.handleOnActionMenu(MenuBarController.java:1269)
        at com.gluonhq.scenebuilder.app@24.0.0-SNAPSHOT/com.oracle.javafx.scenebuilder.app.menubar.MenuBarController.lambda$new$4(MenuBarController.java:1263)
...
Caused by: java.lang.IllegalStateException: javafx.fxml.LoadException: 
.m2/repository/com/gluonhq/scenebuilder/kit/24.0.0-SNAPSHOT/kit-24.0.0-SNAPSHOT.jar!/com/oracle/javafx/scenebuilder/kit/util/control/paintpicker/PaintPicker.fxml:41

        at com.gluonhq.scenebuilder.kit@24.0.0-SNAPSHOT/com.oracle.javafx.scenebuilder.kit.util.control.paintpicker.PaintPicker.<init>(PaintPicker.java:71)
        at com.gluonhq.scenebuilder.kit@24.0.0-SNAPSHOT/com.oracle.javafx.scenebuilder.kit.util.control.paintpicker.PaintPicker.<init>(PaintPicker.java:76)
        at com.gluonhq.scenebuilder.app@24.0.0-SNAPSHOT/com.oracle.javafx.scenebuilder.app.preferences.PreferencesWindowController.controllerDidLoadFxml(PreferencesWindowController.java:209)
        at com.gluonhq.scenebuilder.kit@24.0.0-SNAPSHOT/com.oracle.javafx.scenebuilder.kit.editor.panel.util.AbstractFxmlWindowController.makeRoot(AbstractFxmlWindowController.java:111)
        ... 28 more
Caused by: javafx.fxml.LoadException: 
.m2/repository/com/gluonhq/scenebuilder/kit/24.0.0-SNAPSHOT/kit-24.0.0-SNAPSHOT.jar!/com/oracle/javafx/scenebuilder/kit/util/control/paintpicker/PaintPicker.fxml:41

        at javafx.fxml@23/javafx.fxml.FXMLLoader.constructLoadException(FXMLLoader.java:2727)
...
        at javafx.fxml@23/javafx.fxml.FXMLLoader.loadImpl(FXMLLoader.java:2568)
        at javafx.fxml@23/javafx.fxml.FXMLLoader.load(FXMLLoader.java:2536)
        at com.gluonhq.scenebuilder.kit@24.0.0-SNAPSHOT/com.oracle.javafx.scenebuilder.kit.util.control.paintpicker.PaintPicker.<init>(PaintPicker.java:60)
        ... 31 more
Caused by: java.lang.IllegalAccessException: class javafx.fxml.FXMLLoader$ValueElement (in module javafx.fxml) cannot access a member of class com.oracle.javafx.scenebuilder.kit.util.control.paintpicker.PaintPickerController (in module com.gluonhq.scenebuilder.kit) with package access
...
```

And that indeed explains the root issue:

> class javafx.fxml.FXMLLoader$ValueElement (in module javafx.fxml) cannot access a member of class com.oracle.javafx.scenebuilder.kit.util.control.paintpicker.PaintPickerController (in module com.gluonhq.scenebuilder.kit) with package access

so the problem is indeed accessing the constructor in PaintPickerController that only has package access.

so

--- a/kit/src/main/java/com/oracle/javafx/scenebuilder/kit/util/control/paintpicker/PaintPickerController.java
+++ b/kit/src/main/java/com/oracle/javafx/scenebuilder/kit/util/control/paintpicker/PaintPickerController.java
@@ -80,7 +80,7 @@ public class PaintPickerController {
     public final static RadialGradient DEFAULT_RADIAL
             = new RadialGradient(0.0, 0.0, 0.5, 0.5, 0.5, true, CycleMethod.NO_CYCLE);
 
-    PaintPickerController() {
+    public PaintPickerController() {
         // no-op
     }
 

